src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -841,7 +846,21 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
 
             if (config.TryGetValue("timeout", out value))
             {
-                scriptConfig.FunctionTimeout = TimeSpan.Parse((string)value, CultureInfo.InvariantCulture);
+                TimeSpan requestedTimeout = TimeSpan.Parse((string)value, CultureInfo.InvariantCulture);
+
+                // Only apply limits if this is Dynamic.
+                if (IsDynamicSku() && (requestedTimeout < MinTimeout || requestedTimeout > MaxTimeout))
+                {
+                    string message = $"{nameof(scriptConfig.FunctionTimeout)} must be between {MinTimeout} and {MaxTimeout}.";
+                    throw new ArgumentException(message);
```

Comment:
That's what I had in my original commit and @fabiocav suggested an ArgumentException. I could go either way.

---
In reply to: [78202291](https://github.com/Azure/azure-webjobs-sdk-script/pull/661#discussion_r78202291) [](ancestors = 78202291)